### PR TITLE
Increase fileChunkMaxLength and csvRowsPerQuery to speed up importing of large files

### PR DESCRIPTION
**Pull request by undefz**
Current size of csvRowsPerQuery makes almost impossible to import csv files of large size (personally I find it disturbing starting with 1-2 mb when ping to database is more than 100ms).
Thereby I propose to increase this limit.

### CODE REVIEW

**Comment by Jason-Morcos** [comment]
@undefz Please pull Sequel-Ace dev branch back into your branch so automated tests can be run! :)

**Comment by undefz** [comment]
@Jason-Morcos Done, tests seem to be fine :)

**Comment by Jason-Morcos** [comment]
I think my only potential concern with this PR is that I don't really know the reasoning for why the limits were set at what they previously were. What if we merge this and 1000 CSV rows is too big a chunk size for certain imports?

What if instead we made these values user-configurable (in preferences) and left the defaults at what they currently are?

File: Source/SPDataImport.m [hunk 1]
```diff
@@ -782,8 +782,8 @@ - (void)importCSVFile:(NSString *)filename
 	NSMutableArray *parsedRows = [[NSMutableArray alloc] init];
 	NSMutableArray *parsePositions = [[NSMutableArray alloc] init];
 	NSArray *csvRowArray;
-	NSInteger fileChunkMaxLength = 256 * 1024;
-	NSUInteger csvRowsPerQuery = 50;
+	NSInteger fileChunkMaxLength = 512 * 1024;
+	NSUInteger csvRowsPerQuery = 1000;
```
Comment (Kaspik) [inline Source/SPDataImport.m:786]: This increases the values toooooo much. Afaik Bigger query might be affected in concurrent access and could potentially lead to a deadlock? So if increase, then let's try smaller - like 200 (4x) and not 1000 (20x) :)

**Comment by gboudreau** [comment]
> I think my only potential concern with this PR is that I don't really know the reasoning for why the limits were set at what they previously were. What if we merge this and 1000 CSV rows is too big a chunk size for certain imports?
> 
> What if instead we made these values user-configurable (in preferences) and left the defaults at what they currently are?

Those values were chosen 11 years ago... https://github.com/Sequel-Ace/Sequel-Ace/commit/cc0c0a7842e3bff325fa29c71f5115361981797d
The commit message mentions "CSV rows are grouped into batches of up to 50 (depending on line length)", so it seems to indicate there is code that will limit the number of rows to less than `csvRowsPerQuery`, based on the rows length..? If that is true, one could use a very high value for `csvRowsPerQuery` without big risks.

There is this: https://github.com/Sequel-Ace/Sequel-Ace/blob/dev/Source/SPDataImport.m#L1110
That limits the SQL queries to 250,000 characters. So it will read `csvRowsPerQuery` rows from the CSV (effectively just using more memory, if `csvRowsPerQuery` value is increased), and then import those rows in batches of `csvRowsPerQuery` rows or 250kB (approx), whatever is smaller.

I think inserting rows in batches of 1,000 rows, knowing those INSERTs can never be more than 250k characters, is safe enough nowadays. Even if locking occurs on the server, I can't think of any way this could cause an issue.

I tested with a 38.2MiB .csv, exported from a real table I have (is a mix of POINT, text, enum, int and floats columns).
Approx 120 characters per line, in CSV format. 67 bytes per row, when in MySQL.
Server is on LAN.
- 256kb chunks, 50 rows per INSERT: 152k rows/minute
- 256kB chunks, 200 rows per INSERT: 270k rows/mins
- 512kB chunks, 200 rows per INSERT: same
- 512kB chunks, 1,000 rows per INSERT: 341k rows/mins
- 1MB chunks, 10,000 rows per INSERT: 357k rows/mins
- 1MB chunks, 10,000 rows per INSERT, increase query size limit from 250,000 to 2,500,000: 380k rows/mins
(10,000 rows INSERTs were approx. 1,250,000 characters, so not reaching the limit)
- 1MB chunks, 30,000 rows per INSERT, query size limit 2,500,000: 355k rows/mins

While the CSV was importing, I tried to connect using another session, and SELECT from the same table. No problems there.

I think if we leave the 250,000 characters limit in place, using `csvRowsPerQuery = 1000` is fine, and will indeed more than double the import speed.

**Pull request merged**